sql/core/src/main/scala/org/apache/spark/sql/execution/ShuffledRowRDD.scala

```diff
@@ -154,7 +156,14 @@ class ShuffledRowRDD(
 
   override def compute(split: Partition, context: TaskContext): Iterator[InternalRow] = {
     val shuffledRowPartition = split.asInstanceOf[ShuffledRowRDDPartition]
-    val metrics = context.taskMetrics().createTempShuffleReadMetrics()
+    val tempMetrics = context.taskMetrics().createTempShuffleReadMetrics()
+    // metrics here could be empty cause user can use ShuffledRowRDD directly,
```

Comment:
I don't think we need to consider this case since `ShuffledRowRDD` is a private API. If we do need to consider it, we also need to take care if users pass in a `metrics` that is invalid.

Reply:
do you mean we may leave the `metrics` empty when creating `ShuffledRowRDD` in tests?

Reply:
```
do you mean we may leave the metrics empty when creating ShuffledRowRDD in tests?
```
Yes, like we did in `UnsafeRowSerializerSuite`.
```
I don't think we need to consider this case since ShuffledRowRDD is a private API
```
Got it, after search `new ShuffledRowRDD` in all source code, `UnsafeRowSerializerSuite` is the only place, I'll change the test and delete the default value of `metrics` in this commit.

Reply:
Done in 0348ae5.